pkg/gc/finalizer.go

```diff
@@ -149,29 +146,41 @@ func (f *subscriptionFinalizer) checkGeneration(ctx context.Context, typ core_mo
 			continue
 		}
 
-		old, ok := f.insights[tenantID(tenantId)][typ][key]
-		if !ok || old.id != insight.GetLastSubscription().GetId() || old.generation != insight.GetLastSubscription().GetGeneration() {
-			// something changed since the last check, either subscriptionId or generation were updated
-			// don't finalize the subscription, update map with fresh data
-			f.insights[tenantID(tenantId)][typ][key] = &descriptor{
-				id:         insight.GetLastSubscription().GetId(),
-				generation: insight.GetLastSubscription().GetGeneration(),
+		lastGens := f.insights[tenantID(tenantId)][typ][key]
```

Comment:
Should we rename `f.insights` to `f.onlineInsights`, so it's more obvious we're monitoring only insights that are still online